domain/credential/service/modelcredential.go

```diff
@@ -56,6 +65,7 @@ type CredentialValidationContext struct {
 	ControllerUUID string
 
 	Config         *config.Config
+	MachineState   MachineState
```

Comment:
Gross! Why do we need both service and state? Can we please just one one?

Reply:
We should confirm with Ian, but this domain is just scaffold so it's done on purpose. Again, I only changed the method call here so this domain is not part of the PR.